crates/core_simd/src/first.rs

```diff
@@ -42,6 +47,44 @@ macro_rules! impl_vector {
                     self.0
                 }
             }
+
+            /// Loads a vector from a slice, without doing bounds checking
+            ///
+            /// # Safety
+            /// The length of `src` must be at least `LANES`.
+            #[inline]
+            pub unsafe fn load_from_slice_unchecked(src: &[$type]) -> Self {
+                (src.as_ptr() as *const Self).read_unaligned()
+            }
+
+            /// Loads a vector from a slice.
+            ///
+            /// # Panics
+            /// This function will panic if the length of `src` is not at least `LANES`.
+            #[inline]
+            pub fn load_from_slice(src: &[$type]) -> Self {
+                assert!(src.len() >= LANES, "slice must be at least as long as vector");
+                unsafe { Self::load_from_slice_unchecked(src) }
+            }
```

Comment:
I'm fine with offering an unaligned read as the "default" interface, but panic machinery can often fail to optimize out (and even result in double-retests... a clear compiler error, but one we should be aware of). So my question is, do we want to have the infallible version offer a panicking interface or one that masks and zero-fills?

Reply:
I think panicking is appropriate.  This function will probably rarely be optimal but it's probably necessary to provide and it's no less efficient than zero-filling, I think.  Either way it's just a branch.  In my opinion `from_array` is the infallible version.